ftw/news/tests/test_news_listing_block.py

```diff
@@ -70,3 +72,39 @@ def test_news_listing_block_has_no_rss_link(self, browser):
         browser.login().visit(page)
         self.assertEqual('Subscribe to the RSS feed',
                          browser.find('Subscribe to the RSS feed').text)
+
+    @browsing
+    def test_news_listing_block_shows_lead_image(self, browser):
+        page = create(Builder('sl content page').titled(u'A page'))
+        news_folder = create(Builder('news folder')
+                             .titled(u'News Folder')
+                             .within(page))
+        news = create(Builder('news')
+                      .titled(u'Hello World')
+                      .within(news_folder)
+                      .having(news_date=date(2000, 12, 31)))
+        textblock = create(Builder('sl textblock')
+                           .titled(u'Textblock with image')
+                           .within(news)
+                           .with_dummy_image())
+        utils.create_page_state(news, textblock)
+
+        block = create(Builder('news listing block')
+                       .within(page)
+                       .titled('News listing block')
+                       .having(show_lead_image=True))
+
+        lead_image_css_selector = '.newsListing .newsItem .image img'
+
+        browser.login().visit(page)
+        self.assertEqual(
+            'Textblock with image',
+            browser.css(lead_image_css_selector).first.attrib['title']
+        )
+
+        # Now edit the block so it will not show the lead image.
+        browser.login().visit(block, view='edit')
```

Comment:
You don't need to repeat logging in, it does not change anything.
Having one `login()` is sufficient. :wink: